code/datums/status_effects/debuffs/heart_attack.dm

```diff
@@ -0,0 +1,162 @@
+///Stage two of the heart attack.
+#define ATTACK_STAGE_TWO 100
+///Stage three of the heart attack.
+#define ATTACK_STAGE_THREE 70
+///Stage four of the heart attack.
+#define ATTACK_STAGE_FOUR 35
+///If we reduce heart damage enough, it will recover on its own.
+#define ATTACK_CURE_THRESHOLD 160
+///What is the max oxyloss we're willing to deal, to prevent people from passing out early.
+#define OXYLOSS_MAXIMUM 40
+
+/datum/status_effect/heart_attack
+	id = "heart_attack"
+	status_type = STATUS_EFFECT_UNIQUE
+	remove_on_fullheal = TRUE
+	alert_type = null
+	///A timer that ticks down until the heart fully stops
+	var/time_until_stoppage = 150
+	///Does the victim hear their own heartbeat?
+	var/sound = FALSE
+	///Does this show up on medhuds?
+	var/visible = FALSE
+
+/datum/status_effect/heart_attack/on_apply()
+	var/mob/living/carbon/human/human_owner = owner
+	if(!istype(human_owner) || !human_owner.can_heartattack())
+		return FALSE
+	RegisterSignal(owner, COMSIG_CARBON_LOSE_ORGAN, PROC_REF(end_attack))
+	RegisterSignal(owner, COMSIG_LIVING_MINOR_SHOCK, PROC_REF(minor_shock))
+	RegisterSignal(owner, COMSIG_DEFIBRILLATOR_SHOCKED, PROC_REF(defib_shock))
+	RegisterSignal(owner, COMSIG_LIVING_ELECTROCUTE_ACT, PROC_REF(electrocuted))
+	return TRUE
+
+/datum/status_effect/heart_attack/on_remove()
+	UnregisterSignal(owner, list(COMSIG_CARBON_LOSE_ORGAN, COMSIG_LIVING_MINOR_SHOCK, COMSIG_DEFIBRILLATOR_SHOCKED))
+
+/datum/status_effect/heart_attack/tick()
+	var/mob/living/carbon/human/human_owner = owner
+	if(!istype(human_owner) || !human_owner.can_heartattack())
+		qdel(src) //No heart? No effects.
+		return
+
+	if(time_until_stoppage > ATTACK_CURE_THRESHOLD)
+		owner.visible_message(span_nicegreen("[owner] relaxes [owner.p_their()] body and stops clutching at [owner.p_their()] chest!"), span_nicegreen("The pain in your chest has subsided. You're cured!"))
+		qdel(src)
+		return
+
+	var/oxyloss_sum = 0 //A sum of the oxyloss we will inflict by the end of this cycle.
+
+	if(time_until_stoppage > ATTACK_STAGE_THREE)
+		if(prob(5))
+			owner.playsound_local(owner, 'sound/effects/singlebeat.ogg', 25, FALSE, use_reverb = FALSE)
+			owner.adjustStaminaLoss(5, FALSE)
```

Comment:
Why is this passing `updating_stamina = FALSE`?

Reply:
No clue, probably some copypaste mishap somewhere.